common/test/acceptance/tests/video/test_video_events.py

```diff
@@ -0,0 +1,137 @@
+
+import datetime
+import json
+
+from ..helpers import EventsTestMixin
+from .test_video_module import VideoBaseTest
+
+from common.assertions.events import assert_event_matches
+from opaque_keys.edx.keys import UsageKey, CourseKey
+
+
+class VideoEventsTest(EventsTestMixin, VideoBaseTest):
+    """ Test video player event emission """
+
+    def test_video_control_events(self):
+        """
+        Scenario: Video component is rendered in the LMS in Youtube mode without HTML5 sources
+        Given the course has a Video component in "Youtube" mode
+        And I play the video
+        And I watch 5 seconds of it
+        And I pause the video
+        Then a "load_video" event is emitted
+        And a "play_video" event is emitted
+        And a "pause_video" event is emitted
+        """
+        load_video_promise = self.create_event_of_type_promise('load_video')
+
+        self.navigate_to_video()
+
+        load_video_event = load_video_promise.fulfill()
+
+        self.assert_payload_contains_ids(load_video_event)
+        assert_event_matches({'event_type': 'load_video'}, load_video_event)
+
+        play_video_promise = self.create_event_of_type_promise('play_video')
+        pause_video_promise = self.create_event_of_type_promise('pause_video')
+
+        self.video.click_player_button('play')
+        self.video.wait_for_position('0:05')
+        self.video.click_player_button('pause')
+
+        play_video_event = play_video_promise.fulfill()
+        self.assert_valid_control_event_at_time(play_video_event, 0)
+
+        pause_video_event = pause_video_promise.fulfill()
+        self.assert_valid_control_event_at_time(pause_video_event, self.video.seconds)
+
```

Comment:
My understanding is that one can only create a single promise of type 'play_video' at a time.  So if I want to look at a sequence of play, pause, play, pause, I could create the second pair of promises by cloning the code at 35-36, and inserting it here at 47.  They would then be matched by any events generated after that point, as triggered by subsequent 'click_player_button' calls.  As long as we can wrap the event triggers with an appropriate promise, this approach should work.